File: tests/fixtures/cpp_simple/handler.cpp
#include "service.hpp"
#include "repository.hpp"
#include "models.hpp"
#include <iostream>
#include <vector>

namespace app {

class Handler {
public:
    Handler() : svc_(), repo_() {}

    std::string handleGet(int id) {
        auto result = svc_.getItem(id);
        if (result.empty()) {
            std::cerr << "Item not found: " << id << std::endl;
        }
        return result;
    }

    int handleCreate(const std::string& name) {
        if (name.empty()) {
            throw std::invalid_argument("Name cannot be empty");
        }
        return svc_.createItem(name);
    }

    bool handleDelete(int id) {
        return svc_.deleteItem(id);
    }

    std::vector<ItemRecord> handleList() {
        return svc_.listItems();
    }

    AppConfig getConfig() const {
        return config_;
    }

private:
    DataService svc_;
    ItemRepository repo_;
    AppConfig config_;
};

} // namespace app

int main() {
    app::Handler h;
    int id = h.handleCreate("test");
    std::cout << "Created: " << id << std::endl;
    std::cout << "Get: " << h.handleGet(id) << std::endl;
    auto items = h.handleList();
    std::cout << "Total: " << items.size() << std::endl;
    return 0;
}

File: tests/fixtures/go_simple/handler.go
package main

import (
	"fmt"
	"myapp/service"
	"myapp/middleware"
)

type Handler struct {
	svc *service.DataService
	log *middleware.Logger
}

func NewHandler() *Handler {
	return &Handler{
		svc: service.NewDataService(),
		log: middleware.NewLogger("handler"),
	}
}

func (h *Handler) HandleGet(id int) string {
	h.log.Info("Getting item", id)
	result := h.svc.GetItem(id)
	if result == "" {
		h.log.Warn("Item not found", id)
		return ""
	}
	return result
}

func (h *Handler) HandleCreate(name string) int {
	h.log.Info("Creating item", 0)
	id := h.svc.CreateItem(name)
	h.log.Info("Created item", id)
	return id
}

func (h *Handler) HandleDelete(id int) bool {
	h.log.Info("Deleting item", id)
	return h.svc.DeleteItem(id)
}

func (h *Handler) HandleList() []service.ItemRecord {
	return h.svc.ListItems()
}

func main() {
	h := NewHandler()
	id := h.HandleCreate("test-item")
	fmt.Println("Created:", id)
	fmt.Println("Get:", h.HandleGet(id))
	fmt.Println("All:", h.HandleList())
}
